mlir/lib/Conversion/GPUToLLVMSPV/GPUToLLVMSPV.cpp

```diff
@@ -53,6 +52,17 @@ static LLVM::LLVMFuncOp lookupOrCreateSPIRVFn(Operation *symbolTable,
         symbolTable->getLoc(), name,
         LLVM::LLVMFunctionType::get(resultType, paramTypes));
     func.setCConv(LLVM::cconv::CConv::SPIR_FUNC);
+    func.setNoUnwind(true);
+    func.setWillReturn(true);
```

Comment:
We shouldn't be adding `willreturn` to all operations. Add it only to the builtins, please. `barrier()` and `sub_group_shuffle()` may not return if not all work-items execute the function. You can pass an additional argument to the function.

Reply:
I was thinking that that will only happen when you use the function incorrectly, which in undefined behaviour anyway, but I'm not that sure so that's fine.

Reply:
You're actually right. I wasn't aware `willreturn` took UB into account. Feel free to add back.